addons/analytic/models/analytic_mixin.py

```diff
@@ -176,7 +176,28 @@ def _validate_distribution(self, **kwargs):
                     raise ValidationError(_("One or more lines require a 100% analytic distribution."))
 
     def _sanitize_values(self, vals, decimal_precision):
-        """ Normalize the float of the distribution """
+        """
+        Create analytic distribution from analytic_line_ids at import
+        and normalize the float of the distribution.
+        """
+        def get_analytic_accounts(analytic_line_vals):
+            return self.env['account.analytic.account'].browse([
+                analytic_line_vals[fname]
+                for fname in plan_fnames
+                if fname in analytic_line_vals
+            ])
+
+        if self.env.context.get('import_file') and vals.get('analytic_line_ids'):
+            line_balance = vals['balance']
+            analytic_dist = {}
+            plan_fnames = self.env['account.analytic.line']._get_plan_fnames()
+            for analytic_line in vals['analytic_line_ids']:
+                account_id = ",".join(str(account_id) for account_id in get_analytic_accounts(analytic_line[2]).ids)
+                analytic_line_amount = analytic_line[2]['amount']
```

Comment:
What if `amount` is not set? (i.e. only changing the accounts)

It is a likely operation if we do it on existing posted moves for instance (even if it would be more straightforward from the list view of analytic lines)